storm-server/src/main/java/org/apache/storm/metric/timed/TimerDecorated.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.storm.metric.timed;
+
+import com.codahale.metrics.Timer;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+public interface TimerDecorated extends AutoCloseable {
+    public boolean hasStopped();
+
+    default boolean hasStopped(final AtomicReference<Timer.Context> timing) {
+        return timing.get() == null;
+    }
+
+    public long stopTiming();
+
+    /**
+     * Stop the timer for measured object. Should be executed only once.
+     *
+     * @return Time a object is in use, or under measurement, in nanoseconds.
+     * @throws NullPointerException this method is called more than once.
+     */
+    default long stopTiming(final AtomicReference<Timer.Context> timingRef) throws NullPointerException {
+        Timer.Context timing = timingRef.get();
+        if (timing != null) {
+            long lapses = timing.stop();
+            timingRef.set(null);
```

Comment:
This is potentially a problem, because AtomicReference implies that it is being used from multiple threads.  If that is true, then there is a race here, where you might have multiple threads actually call stop on it at the same time, because the get does not atomically set it to null.

Reply:
I've just changed so to getAndSet(). In actual case there might not be multiple threads calling it. I still need to investigate a bit.